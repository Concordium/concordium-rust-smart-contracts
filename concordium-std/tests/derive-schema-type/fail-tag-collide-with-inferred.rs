//! Ensure `derive(SchemaType)` fails when a 'tag' attribute collides with the
//! default tag for another variant.
use concordium_std::*;

#[derive(SchemaType)]
#[concordium(repr(u8))]
enum MyTaggedEnum {
    One,
    #[concordium(tag = 0)]
    Two,
}

fn main() {}
