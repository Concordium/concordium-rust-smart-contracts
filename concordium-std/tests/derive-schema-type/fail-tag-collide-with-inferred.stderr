error: The tag of 'Two' collides with the tag of 'One'.
 --> tests/derive-schema-type/fail-tag-collide-with-inferred.rs:9:18
  |
9 |     #[concordium(tag = 0)]
  |                  ^^^

error: The tag of 'One' collides with the tag of 'Two'.
 --> tests/derive-schema-type/fail-tag-collide-with-inferred.rs:8:5
  |
8 |     One,
  |     ^^^
